pyterrier/ops.py

```diff
@@ -1,46 +1,38 @@
-from .transformer import Transformer, Estimator, get_transformer, Scalar
+from .transformer import Transformer, Estimator, get_transformer
 from .model import add_ranks
-from matchpy import Operation, Arity
+from collections import deque
 from warnings import warn
-from typing import Iterable
+from typing import Iterable, List, Optional, Protocol, runtime_checkable
+from itertools import chain
 import pandas as pd
 
-class BinaryTransformerBase(Transformer,Operation):
```

Comment:
have we lost something about Arity? is it not a useful concept to keep around? Is there anything that all operators inherit from?

Reply:
I don't see operators as a subclass of transformers. To me, an operator is just syntactic sugar -- they make it easier to combine common transformers.

If the distinction is that an operator is a type of transformers that take other transformers as input, we do not consistently apply a base class as of now. E.g., some of the `pyterrier-caching` transformers take other transformers, as does `GAR`. Probably others too.